sdk/inc/azure/iot/az_iot_hub_client.h

```diff
@@ -55,8 +55,36 @@ typedef struct
    * The model ID used to identify the capabilities of a device based on the Digital Twin document.
    */
   az_span model_id;
+
+  /**
+   * The method_twin_content_type is sent in the username to inform IoT Hub what format the device
+   * expects Twin Document and Direct Method payloads to use. If this option is not set, the default
+   * is JSON. If the content type provided is not supported by IoT Hub, the service will close the
+   * connection with error 4 code indicating Bad Username/Password.
+   */
+  az_span method_twin_content_type;
+
 } az_iot_hub_client_options;
 
+/**
+ * @brief Azure IoT Hub Client option values. Not all options will have pre-defined values.
+ *
+ */
+
+/**
+ * @brief CBOR content type. Sent in username to inform IoT Hub the device expects CBOR for Twin
+ * Document and Direct Method payloads.
+ * @note It can be used by wrapping the macro in an #AZ_SPAN_FROM_STR macro as a parameter.
+ */
+#define AZ_IOT_HUB_CLIENT_OPTION_METHOD_TWIN_CONTENT_TYPE_CBOR "application%2Fcbor"
+
+/**
+ * @brief JSON content type. Sent in username to inform IoT Hub the device expects JSON for Twin
+ * Document and Direct Method payloads.
+ * @note It can be used by wrapping the macro in an #AZ_SPAN_FROM_STR macro as a parameter.
+ */
+#define AZ_IOT_HUB_CLIENT_OPTION_METHOD_TWIN_CONTENT_TYPE_JSON "application%2Fjson"
```

Comment:
I updated to the following.  What do you think?

```
/**
 * @brief Azure IoT Hub Client method_twin_content_type option value: CBOR.
 * @note It can be used by wrapping the macro in an #AZ_SPAN_FROM_STR macro as a parameter.
 */
#define AZ_IOT_HUB_CLIENT_OPTION_METHOD_TWIN_CONTENT_TYPE_CBOR "application%2Fcbor"

/**
 * @brief Azure IoT Hub Client method_twin_content_type option value: JSON.
 * @note It can be used by wrapping the macro in an #AZ_SPAN_FROM_STR macro as a parameter.
 */
#define AZ_IOT_HUB_CLIENT_OPTION_METHOD_TWIN_CONTENT_TYPE_JSON "application%2Fjson"
```